### PR TITLE
C++ examples

### PR DESCRIPTION
**Pull request by yu239**
C++ examples for simple_race, atari, and xworld.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,114 +80,119 @@ To get started, several examples of the simulator Python APIs can be found in
 ```
 <xworld_path>/python/examples
 ```
+And several C++ examples can be found in
+```
+<xworld_path>/examples
+```
+Generally, C++ APIs are more flexible but expose more defails than Python APIs.
```
Comment (skylian) [inline README.md]: add something like, "please refer to .sh files for how to run those examples."

**Comment by skylian** [comment]
there are two versions of gflags out there that use different namespaces. consider this issue as TODO

File: examples/test_atari.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+#include "games/arcade/arcade_simulator.h"
+
+using namespace simulator::arcade_game;
+using namespace simulator;
+
+DECLARE_int32(context);
+
+int main(int argc, char** argv) {
+    gflags::ParseCommandLineFlags(&argc, &argv, true);
```
Comment (skylian) [inline examples/test_atari.cpp]: need to add these codes before this line:
#ifndef GFLAGS_GFLAGS_H_
namespace gflags = google;
#endif

same for other cpp files

Comment (yu239) [reply examples/test_atari.cpp]: This is not easier. We need to add these lines for every file.

File: README.md [hunk 3]
```diff
@@ -80,114 +80,119 @@ To get started, several examples of the simulator Python APIs can be found in
 ```
 <xworld_path>/python/examples
 ```
+And several C++ examples (run the .sh scripts inside) can be found in
+```
+<xworld_path>/examples
+```
+Generally, C++ APIs are more flexible but expose more defails than Python APIs.
```
Comment (Haichao-Zhang) [inline README.md]: defails -> details

Comment (yu239) [reply README.md]: changed.

**Pull request merged**